### DISCUSSION TITLE
Plotting ISCE outputs in MATLAB

### DISCUSSION THREAD

**Discussion by asobouti** [isce-framework/isce2#449] [Q&A] [answered]
Hello,
I am using ISCE to produce sentinel-1 interferograms and I prefer to read the outputs in MATLAB. I've already plotted the complex warped interferograms (filt_topophase.flat.geo) in MATLAB but I can not understand the format of unwrapped one (filt_topophase.unw.geo). Does anyone know the format of ISCE unwrapped interferograms?

**Comment by fdelgadodelapuente** [comment, marked as answer]
Hi,

It is a real*4 two band file. Try the following code


h =fopen('filt_topophase.unw.geo','r');
[F,count] = fread(h,2*nx*ny,'float32');
rmg = reshape(F,2*nx,ny);
phs = rmg(nx+1:nx*2,:);
mag = flipud(rmg(1:nx,:)');
phs = flipud(phs'); 

Here mag and phs are the amplitude and phase of the unwrapped interferogram. You need to get nx and ny from the file metadata

Francisco

**Reply by asobouti** [reply]
Thank you Francisco, it worked!

**Reply by camurad12** [reply]
Hi AlirezaS89, how did you manage to plot the outputs in MATLAB once you read them? I have also been able to read the outputs in MATLAB but haven't been able to properly plot them.

**Reply by asobouti** [reply]
Hi camurad12. Try following codes in MATLAB:

lambda_size = xmlread_as('filt_topophase.flat.geo.xml','coordinate1','size');
phi_size = xmlread_as('filt_topophase.flat.geo.xml','coordinate2','size');
fid = fopen('filt_topophase.flat.geo');
cint = fread(fid,'float32');
fclose(fid);
real_part = cint(1:2:length(cint));
cplx_part = cint(2:2:length(cint));
cint = real_part + 1i*cplx_part;
phase = angle(cint);
phase = reshape(phase,lambda_size,[]);
figure
imagesc(phase')
colormap(flipud(jet)),colorbar
